File: SDBMS/SDBMS_BASE_CLASS.cpp
#include "SDBMS_BASE_CLASS.h"

int SDBMS::SDBMS_BASE_CLASS::maxUsedObjectId = 0;

SDBMS::SDBMS_BASE_CLASS::SDBMS_BASE_CLASS()
{
    objectId = ++maxUsedObjectId;
}


SDBMS::SDBMS_BASE_CLASS::~SDBMS_BASE_CLASS()
{
}

int SDBMS::SDBMS_BASE_CLASS::GetObjectId()
{
    return objectId;
}

File: sdbms_base_class.py

class SdbmsBaseClass(object):
    """ Base class for all sdbms classes """

    # class wide counter
    __object_id = 0

    def __init__(self):
        SdbmsBaseClass.__object_id += 1
        self.__id = SdbmsBaseClass.__object_id

    @property
    def obj_id(self):
        return self.__id
